internal/layer2/announcer.go

```diff
@@ -73,7 +74,10 @@ func (a *Announce) updateInterfaces() {
 			continue
 		}
 		if _, err = os.Stat("/sys/class/net/" + ifi.Name + "/master"); !os.IsNotExist(err) {
-			continue
+			masterOperState, err := ioutil.ReadFile("/sys/class/net/" + ifi.Name + "/master/operstate")
+			if err == nil && strings.Contains(string(masterOperState), "up") {
```

Comment:
worth logging the error if not nil? Seems like it should never happen.

Not related to this PR, but netlinks are easier to handle in go with https://github.com/vishvananda/netlink (which is used by a lot of CNI plugins)